### ISSUE TITLE
A different increasing loss issue

### ISSUE THREAD

**Issue by jeffreykthomas** [karpathy/nanoGPT#269]
Has anyone seen a loss pattern like this? I tried the solutions in issue 167 (disabling fused AdamW), to no avail. I'm running the 'scratch' code after cloning with no modifications, other than only two gpus, so:

$ torchrun --standalone --nproc_per_node=2 train.py config/train_gpt2.py

![image](https://user-images.githubusercontent.com/18131906/236632007-3424bd90-ccd5-43bb-b9cc-0aa3482ad84b.jpeg)

https://github.com/karpathy/nanoGPT/issues/167

**Comment by jeffreykthomas** [comment]
Ps have tried PyTorch 2.0.0 and the nightly 2.1.0, CUDA 11.8, same issue with both. Code was run without any modifications to the last commit, and also ran the code with no modifications after a commit about 2 months ago.

**Comment by jeffreykthomas** [comment]
Changing the dtype to float16 seems to have resolved the issue, at least through 72k iterations. Does that mean it might be a cuda or hardware issue?

**Comment by nitinvetcha** [comment]
Hi, 
I am having the same issue 
Could you please confirm if changing the dtype to float16, has resolved the issue after 72k iterations as well ?
